### PR TITLE
V5 audioplayer2

### PR DESCRIPTION
**Pull request by ryanfrancesconi**
🤺

### CODE REVIEW

**Comment by ryanfrancesconi** [comment]
Here's a fork of Cookbook changed to this player, just for an example. Uses an extension to support the schedule methods.

https://github.com/ryanfrancesconi/Cookbook/tree/audioplayer2

File: Tests/AudioKitTests/Node Tests/Generator Tests/AudioPlayer2Tests+Realtime.swift [hunk 1]
```diff
@@ -0,0 +1,198 @@
+import AudioKit
+import AVFoundation
+import CAudioKit
+import XCTest
+
+// Real time functions, for local testing only.
+// These simulate a user interacting with the player via an UI
+// Real time functions, for local testing only
+
+// Thse are organized like this so they're easy to comment out
```
Comment (wtholliday) [inline Tests/AudioKitTests/Node Tests/Generator Tests/AudioPlayer2Tests+Realtime.swift:10]: I think you can disable individual tests via Xcode.

File: Tests/AudioKitTests/Node Tests/Generator Tests/AudioPlayer2Tests+Realtime.swift [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+            return
+        }
+        let engine = AudioEngine()
+        engine.output = player
+        try? engine.start()
+
+        player.completionHandler = { Log("🏁 Completion Handler") }
+        var duration = player.duration
+
+        Log("▶️")
```
Comment (wtholliday) [inline Tests/AudioKitTests/Node Tests/Generator Tests/AudioPlayer2Tests+Realtime.swift:46]: I dig your emojis. I started doing that in my own code.

Comment (ryanfrancesconi) [reply Tests/AudioKitTests/Node Tests/Generator Tests/AudioPlayer2Tests+Realtime.swift:46]: 🎃

File: Tests/AudioKitTests/Node Tests/Generator Tests/AudioPlayer2Tests+Realtime.swift (hunk 1 above)
Comment (ryanfrancesconi) [reply Tests/AudioKitTests/Node Tests/Generator Tests/AudioPlayer2Tests+Realtime.swift:10]: oh, so my thought there was avoiding having CI do those tests. is there a way to control that?

**Comment by wtholliday** [comment]
Looks good to me. Are you planning to do fade in/out in a subsequent change?

**Comment by ryanfrancesconi** [comment]
Definitely could. I'm just not sure what things are desired and what aren't. I don't want to overdo it like i did last time.

File: Tests/AudioKitTests/Node Tests/Generator Tests/AudioPlayer2Tests+Realtime.swift (hunk 1 above)
Comment (wtholliday) [reply Tests/AudioKitTests/Node Tests/Generator Tests/AudioPlayer2Tests+Realtime.swift:10]: I was thinking this:

image

**Comment by aure** [comment]
I think you can go ahead and make this be "AudioPlayer" right in this PR. Involves deleting some files, renaming some, etc.

File: Tests/AudioKitTests/Node Tests/Generator Tests/AudioPlayer2Tests+Realtime.swift (hunk 1 above)
Comment (aure) [reply Tests/AudioKitTests/Node Tests/Generator Tests/AudioPlayer2Tests+Realtime.swift:10]: Where does it save this info if we no longer have an Xcode project file?

**Comment by ryanfrancesconi** [comment]
> I think you can go ahead and make this be "AudioPlayer" right in this PR. Involves deleting some files, renaming some, etc.

you want me to remove the old AudioPlayer?

**Pull request merged**